protocol/messenger_backup.go

```diff
@@ -220,3 +299,54 @@ func (m *Messenger) syncBackupContact(ctx context.Context, contact *Contact) *pr
 		TrustStatus:        int64(contact.TrustStatus),
 	}
 }
+
+func (m *Messenger) backupProfile(ctx context.Context, clock uint64) ([]*protobuf.Backup, error) {
+	displayName, err := m.settings.DisplayName()
+	if err != nil {
+		return nil, err
+	}
+
+	displayNameClock, err := m.settings.GetSettingLastSynced(settings.DisplayName)
+	if err != nil {
+		return nil, err
+	}
+	if displayNameClock == 0 {
+		displayNameClock = clock
```

Comment:
This I am not sure, why are we setting the displayNameClock to a non-zero value?

Reply:
>  I didn't want to initialize settings_sync_clock in that case (even it should be by my opinion) cause I was not sure about all use cases for syncing among devices

I think you are right here, this should be the way, `CreateSettings` should set the sync clock value of any field that's user set, that would be the root cause of the issue, since the above code might override a user display_name, since it might have changed it on a different device before this code is triggered, but before the backup is received.

We can address  it in a separate PR if you'd like, but I think we should fix the root cause, since the code above I believe might introduce a race condition/bug.
If you decide that we should address it in a separate PR, would you mind creating an issue describing the issue so we can track it please?

Reply:
Handled here in this PR.